tests/test_parallel.py

```diff
@@ -54,48 +47,53 @@ def mock_child_job():
     return Response(dict(params=params, arguments=args, **_MOCK_JOB_KWARGS))
 
 
+@pytest.mark.parametrize(
+    "num_failures,max_submit_retries,submit_call_count,should_fail,from_template_id",
+    [
+        # Test that submission doesn't fail when there are no mock API errors.
+        (0, 0, 3, False, None),
+        (0, 5, 3, False, None),
+        # Test that submission fails when there are API errors and too few retries.
+        (2, 0, 1, True, None),
+        (2, 1, 1, True, None),
+        (2, 2, 2, True, None),
+        # Test that submission doesn't fail when there are mock API errors and
+        # sufficient retries.
+        (2, 3, 5, False, None),
+        (2, 4, 5, False, None),
+        # Using a template.
+        (0, 5, 3, False, 13),
+        (1, 5, 4, False, 13),
+        (1, 0, 1, True, 13),
+    ],
+)
+@mock.patch("civis.parallel._robust_pickle_download")
+@mock.patch("civis.io._files.requests")
 @mock.patch("civis.futures.time.sleep", side_effect=lambda x: None)
-def test_retries(m_sleep):
-    """Make sure that job submission retry behavior works."""
-
-    # Test that submission doesn't fail when there are no mock API errors.
-    _test_retries_helper(0, 0, False, None)
-    _test_retries_helper(0, 5, False, None)
-
-    # Test that submission fails when there are API errors and too few retries.
-    _test_retries_helper(2, 0, True, None)
-    _test_retries_helper(2, 1, True, None)
-    _test_retries_helper(2, 2, True, None)
-
-    # Test that submission doesn't fail when there are mock API errors and
-    # sufficient retries.
-    _test_retries_helper(2, 3, False, None)
-    _test_retries_helper(2, 4, False, None)
-
-
-def test_from_template_id():
-    _test_retries_helper(0, 5, False, 13)
-    _test_retries_helper(1, 5, False, 13)
-    _test_retries_helper(1, 0, True, 13)
-
-
 @mock.patch.object(civis.parallel, "_ContainerShellExecutor")
 @mock.patch.object(civis.parallel, "CustomScriptExecutor")
-@mock.patch.object(civis.parallel, "_CivisBackendResult")
```

Comment:
The internals of `joblib.Parallel` have changed substantially between v1.2.0 and v1.3.0. Here, we can no longer mock out `_CivisBackendResult`, as we need to exercise more real, un-mocked `civis.parallel` code than before. So most of the changes in this test are really about refactoring the mocking to the "inner" layer of a `civis.APIClient` instance.

Also switching to `@pytest.mark.parametrize` to run all test cases in case some of them might fail.